app/helpers/auto-approval-check.ts

```diff
@@ -0,0 +1,133 @@
+import { Quota } from '@prisma/client';
+import _each from 'lodash-es/each';
+import _flatMap from 'lodash-es/flatMap';
+import { defaultCpuOptionsLookup, defaultMemoryOptionsLookup, defaultStorageOptionsLookup } from '@/../app/constants';
+import { ResourceType, totalMetrics } from '@/services/openshift-kubernetis-metrics/helpers';
+import getPodMetrics from '@/services/openshift-kubernetis-metrics/pod-usage-metrics';
+import { extractNumbers } from '@/utils/string';
+
+interface Quotas {
+  testQuota: Quota;
+  toolsQuota: Quota;
+  developmentQuota: Quota;
+  productionQuota: Quota;
+}
+
+// check if request contains quota change
+export const checkIfNoQuotaChange = (currentQuota: Quotas, requestedQuota: Quotas) => {
+  // @ts-ignore
+  return _each(currentQuota, (quota: Quota, envQuota: keyof Quotas) => {
+    // @ts-ignore
+    return _each(quota, (resource: string, resourceName: keyof Quota) => {
+      const currentVal = resource;
+      const requestedVal = requestedQuota[envQuota][resourceName];
+      return extractNumbers(requestedVal)[0] === extractNumbers(currentVal)[0];
+    });
+  });
+};
+
+// if quota was changed check if it was undngrade quota request
+// TODO replace isQuotaUpgrade at app/emails/_templates/private-cloud/TeamEditRequest.tsx
+export const checkIfQuotaUpgrade = (currentQuota: Quotas, requestedQuota: Quotas) => {
+  // @ts-ignore
+  const isIncreased = _each(currentQuota, (quota: Quota, envQuota: keyof Quotas) => {
+    // @ts-ignore
+    return _each(quota, (resource: string, resourceName: keyof Quota) => {
+      const currentVal = resource;
+      const requestedVal = requestedQuota[envQuota][resourceName];
+      return extractNumbers(requestedVal)[0] > extractNumbers(currentVal)[0];
+    });
+  });
+  return isIncreased;
+};
+
+const resourceOrders = {
+  cpu: defaultCpuOptionsLookup,
+  memory: defaultMemoryOptionsLookup,
+  storage: defaultStorageOptionsLookup,
+};
+
+// Helper function to check resource utilization
+const checkUtilization = async (
+  requestedQuota: Quotas,
+  licencePlate: string,
+  cluster: string,
+  namespace: string,
+  resource: ResourceType,
+) => {
+  const podMetricsData = await getPodMetrics(licencePlate, namespace, cluster);
+  if (!podMetricsData) return false;
+
+  const { totalUsage, totalLimit } = totalMetrics(podMetricsData, resource);
```

Comment:
1. rename to `getTotalMetrics`.
2. use the limit value form the database instead of `totalLimit`, which is the sum of the deployed pods.